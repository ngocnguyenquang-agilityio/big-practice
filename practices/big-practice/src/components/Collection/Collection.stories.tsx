// Types
import type { Meta, StoryObj } from '@storybook/react';

// Components
import Collection from '.';
import { COLLECTION_TYPE } from '@interfaces';

const mockListCollection = [
  { value: 'all', label: 'All' },
  { value: 'phones', label: 'Phones' },
];

const meta = {
  title: 'Components/Collection',
  component: Collection,
} satisfies Meta<typeof Collection>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    list: mockListCollection,
    title: 'Collection',
    type: COLLECTION_TYPE.CATEGORY,
  },
};
