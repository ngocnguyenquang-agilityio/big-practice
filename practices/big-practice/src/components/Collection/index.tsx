// Libs
import { Link } from 'react-router-dom';

// Types
import { ICategoryList } from '@interfaces';

// Helpers
import { isEmpty } from '@helpers/utils';

const Collection = ({ categories = [], selectingItem }: ICategoryList): JSX.Element => {
  return (
    <nav>
      <h3 className='text-xs text-neutral-400 md:block'>Category</h3>
      <ul
        className='block'
        data-testid='collection'
      >
        {categories.map((item) => (
          <li
            className={`mt-2 flex text-white ${selectingItem === item.value ? 'font-bold underline' : ''}`}
            key={item.value}
          >
            <Link
              to={isEmpty(item.value) ? '/' : `/category/${item.value}`}
              className='w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100'
            >
              {item.label}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Collection;
