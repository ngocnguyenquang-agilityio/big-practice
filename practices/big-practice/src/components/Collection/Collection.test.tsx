import { screen, render } from '@testing-library/react';

import Collection from '.';

const mockListCollection = ['All', 'Hoodie', 'Jacket', 'Shirt']

describe('Collection component', () => {
  test('Should render correctly', () => {
    const comp = render(<Collection list={mockListCollection} />)
    expect(comp).toBeTruthy()
  })

  test('Should match snapshot', () => {
    const comp = render(<Collection list={mockListCollection} />)
    expect(comp).toMatchSnapshot()
  })

  test('Should render correct number of items', async () => {
    render(<Collection list={mockListCollection} />)
    const items = await screen.getByTestId('collection')
    expect(items.children).toHaveLength(4)
  })
})
