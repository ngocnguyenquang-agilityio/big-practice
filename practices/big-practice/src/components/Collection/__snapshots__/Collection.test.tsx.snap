// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h3
          class="text-xs text-neutral-400 md:block"
        >
          Collection
        </h3>
        <ul
          class="block"
          data-testid="collection"
        >
          <li
            class="mt-2 flex text-white "
          >
            <a
              class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
              href="/category/phones"
            >
              Phone
            </a>
          </li>
          <li
            class="mt-2 flex text-white "
          >
            <a
              class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
              href="/category/laptops"
            >
              Laptop
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h3
        class="text-xs text-neutral-400 md:block"
      >
        Collection
      </h3>
      <ul
        class="block"
        data-testid="collection"
      >
        <li
          class="mt-2 flex text-white "
        >
          <a
            class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
            href="/category/phones"
          >
            Phone
          </a>
        </li>
        <li
          class="mt-2 flex text-white "
        >
          <a
            class="w-full text-sm underline-offset-4 hover:underline hover:text-neutral-100"
            href="/category/laptops"
          >
            Laptop
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
